### PR TITLE
Sbachmei/mic 5668/bugfix handle empty years for means

### PR DESCRIPTION
**Pull request by stevebachmeier**
### Description

- *Category*:  bugfix, test
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5668

### Changes and notes
 
This implements a few of the (currently skipped) core.py integration tests
as (slow-running) unit tests (temporarily). In doing this, I found there is a bug
in how we are handling `years="all", data_type="means"`. 

### Testing

All tests pass. NOTE: I'm skipping all relative_risk and paf parameters
since that takes SO long and a followup PR will use mocked data anyway.
i.e. I'm assuming there's nothing special about the location/year return
for rr and paf data.

### CODE REVIEW

File: src/vivarium_inputs/core.py [hunk 1]
```diff
@@ -176,8 +176,8 @@ def get_raw_incidence_rate(
     data = utilities.filter_data_by_restrictions(
         data, restrictions_entity, "yld", utility_data.get_age_group_ids()
     )
-    data = utilities.normalize(data, data_type.value_columns, fill_value=0)
     data = data.filter(DEMOGRAPHIC_COLUMNS + data_type.value_columns)
+    data = utilities.normalize(data, data_type.value_columns, fill_value=0)
```
Comment (stevebachmeier) [inline src/vivarium_inputs/core.py:180]: In multiple places throughout this PR you'll see I moved the filtering of columns _prior_ to the normalizing. I do this b/c the normalizing includes various groupbys and filtering beforehand provides a very large speedup

File: tests/unit/test_core.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
+import pytest
```
Comment (stevebachmeier) [inline tests/unit/test_core.py:1]: I strongly urge you to ignore the first commit when reviewing (which does nothing more than copies this from the TESTS_TO_IMPLEMENT folder)

File: tests/unit/test_core.py [hunk 3]
```diff
@@ -92,22 +88,18 @@ def check_year_in_data(entity, measure, location, years):
 
 
 @pytest.mark.parametrize("entity_details", ENTITIES_C, ids=lambda x: x[0].name)
-@pytest.mark.parametrize("measure", MEASURES_C, ids=lambda x: x[0])
-@pytest.mark.parametrize("location", LOCATIONS_C)
-def test_core_causelike(entity_details, measure, location):
```
Comment (stevebachmeier) [inline tests/unit/test_core.py]: throughout this module, I've deleted all of the `test_core` functions. That is already covered by `test_get_measure.py`.

File: tests/unit/test_core.py [hunk 4]
```diff
@@ -188,69 +158,95 @@ def test_slow_year_id_risklike(location):
...
-def test_pulling_multiple_locations(entity_details, measure, locations):
+def test_multiple_locations_causelike(entity_details, measure, locations, data_type):
+    year = 2021
+    location_id_mapping = {
+        "Ethiopia": 179,
+        "Nigeria": 214,
+    }
     entity, entity_expected_measures = entity_details
-    measure_name, measure_id = measure
-    tester = success_expected if (entity_expected_measures & measure_id) else fail_expected
```
Comment (stevebachmeier) [inline tests/unit/test_core.py]: I didn't like keeping `success_expected` and `fail_expected` around just for this one test so I implemented the raise vs return logic directly

**Review by hussain-jafari — APPROVED** [review]

**Review by albrja — APPROVED** [review]

**Review by rmudambi — APPROVED** [review]
Looking forward to the speed-ups!

**Pull request merged**